### PR TITLE
app-admin/ngxtop: migrate to PEP517 build

### PR DESCRIPTION
**Pull request by hydrapolic**
Tested on ~amd64, thanks!

### CODE REVIEW

File: app-admin/ngxtop/ngxtop-0.0.3_pre141201-r3.ebuild [hunk 1]
```diff
@@ -1,11 +1,11 @@
-# Copyright 1999-2023 Gentoo Authors
+# Copyright 1999-2024 Gentoo Authors
 # Distributed under the terms of the GNU General Public License v2
 
 EAPI=8
 
-PYTHON_COMPAT=( python3_{9..11} )
+PYTHON_COMPAT=( python3_{9..12} )
```
Comment (mgorny) [inline app-admin/ngxtop/ngxtop-0.0.3_pre141201-r3.ebuild:6]: `OldPythonCompat`.

File: app-admin/ngxtop/ngxtop-0.0.3_pre141201-r3.ebuild [hunk 2]
```diff
@@ -1,11 +1,11 @@
...
 PYTHON_REQ_USE="sqlite"
-
+DISTUTILS_USE_PEP517=setuptools
```
Comment (mgorny) [inline app-admin/ngxtop/ngxtop-0.0.3_pre141201-r3.ebuild:8]: Please keep it in the same block as `PYTHON*` stuff.

Also, while at it, could you check if we can remove `DEPEND` below? It seems unnecessary.

**Review by mgorny — CHANGES_REQUESTED** [review]

File: app-admin/ngxtop/ngxtop-0.0.3_pre141201-r3.ebuild (hunk 2 above)
Comment (hydrapolic) [reply app-admin/ngxtop/ngxtop-0.0.3_pre141201-r3.ebuild:6]: It's needed for the tests, so added a condition to it.

**Comment by hydrapolic** [comment]
Thank you for the review @mgorny, pushed an update.

File: app-admin/ngxtop/ngxtop-0.0.3_pre141201-r3.ebuild [hunk 3]
```diff
@@ -16,12 +17,13 @@ SRC_URI="https://dev.gentoo.org/~jlec/distfiles/${P}.tar.xz"
...
+IUSE="test"
+RESTRICT="!test? ( test )"
 
-RDEPEND="
-	dev-python/docopt[${PYTHON_USEDEP}]
+RDEPEND="dev-python/docopt[${PYTHON_USEDEP}]
 	dev-python/pyparsing[${PYTHON_USEDEP}]
 	dev-python/tabulate[${PYTHON_USEDEP}]"
-DEPEND="${RDEPEND}"
+DEPEND="test? ( ${RDEPEND} )"
```
Comment (mgorny) [inline app-admin/ngxtop/ngxtop-0.0.3_pre141201-r3.ebuild:26]: `distutils_enable_tests` already does that.

Comment (hydrapolic) [reply app-admin/ngxtop/ngxtop-0.0.3_pre141201-r3.ebuild]: Thanks @mgorny, updated the PR.

**Review by mgorny — APPROVED** [review]
Thanks!

**Comment by hydrapolic** [comment]
> Thanks!

Thank you @mgorny !

**Pull request closed without merging**